frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -72,6 +76,15 @@ export function ManagePublicKey() {
                 sender: sender,
             });
         } catch (e: any) {
+            trackAppInsightEvent(featureEvent, {
+                fileUpload: {
+                    status: `Error: ${e.toString()}`,
+                    fileName: file?.name,
```

Comment:
Since the file is being set in `useState` within the same function, and `useState` is asynchronous, I'm not sure this is a reliable way to read the file.